security/pkg/nodeagent/sds/server.go

```diff
@@ -163,6 +177,47 @@ func NewPlugins(in []string) []plugin.Plugin {
 	return plugins
 }
 
+func (s *Server) initDebugServer() {
+	mux := http.NewServeMux()
+	mux.HandleFunc(fmt.Sprintf("%s/sds/workload", debugBase), s.sdsWorkloadDebug)
+	mux.HandleFunc(fmt.Sprintf("%s/sds/gateway", debugBase), s.sdsGatewayDebug)
+	s.debugServer = &http.Server{
+		Addr:    fmt.Sprintf("127.0.0.1:%d", debugPort),
+		Handler: mux,
+	}
+
+	go func() {
+		err := s.debugServer.ListenAndServe()
+		log.Errorf("debug server failure: %s", err)
+	}()
+}
+
+func (s *Server) sdsWorkloadDebug(w http.ResponseWriter, req *http.Request) {
```

Comment:
seems duplicated logic in below ```sdsGatewayDebug```, could you refactoring a little(extract a common func) ?